org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java

```diff
@@ -100,6 +100,18 @@ public boolean applies(IProgressMonitor monitor) throws OperationCanceledExcepti
 		return !directories.isEmpty();
 	}
 
+	@Override
+	public boolean applies(Collection<IPath> buildFiles, IProgressMonitor monitor) {
```

Comment:
In the another `applies`, it will check if the importer is enabled. Looks like you should add the same check in the new `applies` as well. Same for GradleProjectImporter.
 ```
PreferenceManager preferencesManager = JavaLanguageServerPlugin.getPreferencesManager();
if (preferencesManager != null && !preferencesManager.getPreferences().isImportMavenEnabled()) {
	return false;
}
```